### ISSUE TITLE
Segfault with --compile=min and C libraries

### ISSUE THREAD

**Issue by mlhetland** [JuliaLang/julia#30504]
I'm getting segfaults when using `--compile=min`, and it seems related to calling functions from C libraries. For example:

```
$ julia --compile=min
…
julia> using Gumbo
julia> parsehtml("x")
signal (11): Segmentation fault: 11
in expression starting at no file:0
jl_pointerref at /Users/osx/buildbot/slave/package_osx64/build/src/runtime_intrinsics.c:36
do_call at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:324
eval_stmt_value at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:363 [inlined]
eval_body at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:682
jl_interpret_call_callback at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:777
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x10f9fb88f)
unknown function (ip: 0x17)
jl_fptr_interpret_call at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:785
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1831
do_call at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:324
eval_body at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:553
jl_interpret_call_callback at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:777
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x10ffeae0f)
unknown function (ip: 0x13)
jl_fptr_interpret_call at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:785
gumbo_to_jl at /Users/mlh/.julia/packages/Gumbo/G7Qbw/src/conversion.jl:59
gumbo_to_jl at /Users/mlh/.julia/packages/Gumbo/G7Qbw/src/conversion.jl:72 [inlined]
document_from_gumbo at /Users/mlh/.julia/packages/Gumbo/G7Qbw/src/conversion.jl:102
#parsehtml#4 at /Users/mlh/.julia/packages/Gumbo/G7Qbw/src/conversion.jl:13
unknown function (ip: 0x11b3a8e8e)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1831
do_call at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:324
eval_stmt_value at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:363 [inlined]
eval_body at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:682
jl_interpret_call_callback at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:777
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x10e79a18f)
unknown function (ip: 0xffffffffffffffff)
jl_fptr_interpret_call at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:785
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1831
do_call at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:324
eval_stmt_value at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:363 [inlined]
eval_body at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:682
jl_interpret_toplevel_thunk_callback at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:806
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x10e7db11f)
unknown function (ip: 0xffffffffffffffff)
jl_interpret_toplevel_thunk at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:815
jl_toplevel_eval_flex at /Users/osx/buildbot/slave/package_osx64/build/src/toplevel.c:805
jl_toplevel_eval_in at /Users/osx/buildbot/slave/package_osx64/build/src/builtins.c:622
eval at ./boot.jl:319
eval_user_input at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:85
macro expansion at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:117 [inlined]
#28 at ./task.jl:259
jl_apply at /Users/osx/buildbot/slave/package_osx64/build/src/./julia.h:1537 [inlined]
start_task at /Users/osx/buildbot/slave/package_osx64/build/src/task.c:268
Allocations: 766295 (Pool: 766043; Big: 252); GC: 0
```

I get similar problems with JuMP, for example (and when running things as scripts/programs rather than in the REPL).

Context:
```
julia> versioninfo()
Julia Version 1.0.3
Commit 099e826241 (2018-12-18 01:34 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin14.5.0)
  CPU: Intel(R) Core(TM) M-5Y71 CPU @ 1.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, broadwell)
Environment:
  JULIA_VERSION = 1.0
```

**Comment by JeffBezanson** [comment]
This seems to be fixed on master, probably by 7638cb39461481ab5d25478bdf3a7ec6972bfad5 and e2fb8d67f05c40c45f1e414e63081018ccd53d75.

**Comment by mlhetland** [comment]
Right! Haven't tried (not compiling Julia myself at the moment), but no need to keep the issue alive on my account; feel free to close it when you're satisfied it works!
